binding/rust/src/porcupine.rs

```diff
@@ -0,0 +1,383 @@
+
+    use crate::porcupine::Porcupine;
+    use crate::util::{pv_keyword_paths, pv_library_path, pv_model_path};
+
+    #[test]
+    fn test_process() {
+        let basedir_path = "../../";
+        let porcupine = Porcupine::new(
+            pv_library_path(basedir_path.clone()),
+            pv_model_path(basedir_path.clone()),
+            &[PathBuf::from(
+                pv_keyword_paths(basedir_path.clone())
+                    .get("porcupine")
+                    .unwrap(),
+            )],
+            &[0.5],
+        )
+        .expect("Unable to create Porcupine");
+
+        let soundfile =
+            BufReader::new(File::open("../../resources/audio_samples/porcupine.wav").unwrap());
+        let source = Decoder::new(soundfile).unwrap();
+
+        assert_eq!(porcupine.sample_rate(), source.sample_rate());
+
+        let mut results = Vec::new();
+        for frame in &source.chunks(porcupine.frame_length() as usize) {
+            let frame = frame.collect_vec();
+            if frame.len() == porcupine.frame_length() as usize {
+                let keyword_index = porcupine.process(&frame).unwrap();
+                if keyword_index >= 0 {
+                    results.push(keyword_index);
+                }
+            }
+        }
+
+        assert_eq!(results, [0]);
+    }
+
+    #[test]
+    fn test_process_multiple() {
+        let basedir_path = "../../";
+        const KEYWORDS: [&str; 8] = [
+            "americano",
+            "blueberry",
+            "bumblebee",
+            "grapefruit",
+            "grasshopper",
+            "picovoice",
+            "porcupine",
+            "terminator",
+        ];
+
+        let keyword_paths = pv_keyword_paths(basedir_path.clone());
+        let selected_keyword_paths = KEYWORDS
+            .iter()
+            .map(|keyword| PathBuf::from(keyword_paths.get(&keyword.to_string()).unwrap()))
+            .collect::<Vec<_>>();
+
+        let porcupine = Porcupine::new(
+            pv_library_path(basedir_path.clone()),
+            pv_model_path(basedir_path.clone()),
+            &selected_keyword_paths,
+            &[0.5; KEYWORDS.len()],
+        )
+        .expect("Unable to create Porcupine");
+
+        let soundfile = BufReader::new(
+            File::open("../../resources/audio_samples/multiple_keywords.wav").unwrap(),
+        );
+        let source = Decoder::new(soundfile).unwrap();
+
+        assert_eq!(porcupine.sample_rate(), source.sample_rate());
+
+        let mut results = Vec::new();
+        for frame in &source.chunks(porcupine.frame_length() as usize) {
+            let frame = frame.collect_vec();
+            if frame.len() == porcupine.frame_length() as usize {
+                let keyword_index = porcupine.process(&frame).unwrap();
+                if keyword_index >= 0 {
+                    results.push(keyword_index);
+                }
+            }
+        }
+
+        assert_eq!(results, [6, 0, 1, 2, 3, 4, 5, 6, 7]);
```

Comment:
It's better to have a list of keyword strings (e.g. ["porcupine", "alexa"...]). Check that list size against results, and then loop through results and check each result value against the input keyword list and the expected keyword list.  If the string gained from both don't match, then we fail.